### ISSUE TITLE
Two new versions found in project

### ISSUE THREAD

**Issue by marcingrzejszczak** [4finance/uptodate-gradle-plugin#13]
https://travis-ci.org/4finance/micro-infra-spring/builds/34382529

```
:uptodate
New versions available:
'org.springframework:spring-test:4.0.7.RELEASE'
'org.springframework:spring-test:4.1.0.RELEASE'
BUILD SUCCESSFUL
```

**Comment by djKianoosh** [comment]
I think I see a similar, possibly related issue: 

If I have 
```providedCompile 'javax.servlet.jsp:jsp-api:2.2.1-b03'```
I get:
```:uptodate
New versions available:
'javax.servlet.jsp:jsp-api:2.2'
```

but if I have the reverse, 
```providedCompile 'javax.servlet.jsp:jsp-api:2.2'```
then I get:
```:uptodate
New versions available:
'javax.servlet.jsp:jsp-api:2.2.1-b03'
```

**Comment by marcingrzejszczak** [comment]
Hi!

Actually I don't think that the two are related.

In your case there is this beta version that is messing around. Here (https://github.com/4finance/uptodate-gradle-plugin/blob/master/src/main/groovy/com/ofg/uptodate/VersionPatterns.groovy#L6) you can see the patterns that we by default assume to be BETA. You can add this sort of a pattern to the list of excludes:

```
uptodate {
    addExcludedVersionPatterns '.*-demo-?\\d*$'
}
```

But still there is an issue here :) The issue is that if you're using `2.2.1-b03` then we should assume that even if we can't parse it (the bugfix version) then still it's something newer than `2.2`. I'll try to see what went wrong here ;)

BTW If you are using this lib can you sign in here - https://github.com/4finance/uptodate-gradle-plugin/wiki/Testimonials ? :)

**Comment by marcingrzejszczak** [comment]
BTW do you have a project at Github that you ran our lib against?

**Comment by djKianoosh** [comment]
not using it yet, but I created a gist for it in case you needed something basic:
https://gist.github.com/djKianoosh/3bbf1aff00e723c9f6b2

If I do use it on one of my projects I'll make sure to come back and update the wiki.

**Comment by marcingrzejszczak** [comment]
Fixed in version 1.1.1. Should be there in JCenter already :) 

https://github.com/4finance/uptodate-gradle-plugin/blob/master/CHANGELOG.md#111

Can you pull it, check it out and then I would close the ticket?

**Comment by djKianoosh** [comment]
works like a charm. :)
